client/history/caching_redirector.go

```diff
@@ -57,25 +60,46 @@ type (
 		}
 
 		connections            connectionPool
+		goros                  goro.Group
 		historyServiceResolver membership.ServiceResolver
 		logger                 log.Logger
+		membershipUpdateCh     chan *membership.ChangedEvent
```

Comment:
does this have to be a field of the struct? it's only used by the goroutine itself. and tests, but maybe there's another way to inject an event?

Reply:
I'd rather leave it, anything else will be awkward for not much of a benefit.